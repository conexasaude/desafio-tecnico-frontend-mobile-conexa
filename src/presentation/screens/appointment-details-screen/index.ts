export * from './appointment-details-screen';
