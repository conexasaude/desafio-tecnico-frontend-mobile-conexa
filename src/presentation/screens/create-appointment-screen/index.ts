export * from './create-appointment-screen';
