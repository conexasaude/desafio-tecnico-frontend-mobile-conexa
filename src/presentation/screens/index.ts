export * from './appointments-screen';
export * from './create-appointment-screen';
export * from './login-screen';
