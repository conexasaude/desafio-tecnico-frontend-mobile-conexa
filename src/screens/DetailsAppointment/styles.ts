import styled from '@emotion/native';

export const SafeArea = styled.SafeAreaView({
  flex: 1,
});

export const Container = styled.View(({ theme }) => ({
  flex: 1,
  paddingTop: theme.effects.spacing.hg,
  paddingHorizontal: theme.effects.spacing.md,
}));

export const Label = styled.Text(({ theme }) => ({
  fontFamily: theme.fonts.family.bold,
  fontSize: theme.fonts.size.lg,
  color: theme.colors.primary,
}));

export const Description = styled.Text(({ theme }) => ({
  fontFamily: theme.fonts.family.semiBold,
  fontSize: theme.fonts.size.md,
  color: theme.colors.gray[600],
  marginBottom: theme.effects.spacing.md,
}));
