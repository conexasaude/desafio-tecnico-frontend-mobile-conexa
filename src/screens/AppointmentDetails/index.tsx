import { View } from 'react-native';
import { NativeStackScreenProps } from "@react-navigation/native-stack";
import { FontAwesome } from '@expo/vector-icons';
import RootStackParamList from '../../types/rootStackParamList';
import formatsDate from '../../utils/formatsDate';
import { Container, Content, Field, FieldContainer, Header, FieldText, Label } from './styles';
import { useEffect } from 'react';

type Props = NativeStackScreenProps<RootStackParamList, 'AppointmentDetails'>;

export default function AppointmentDetails({ route, navigation }: Props) {
	const { appointment } = route.params;
	
	useEffect(() => {
		navigation.setOptions({ title: `Consulta ${appointment.id}` })
	}, [])

	return (
		<Container>
			<Header>
				<View>
					<FontAwesome name="user-circle-o" size={72} color="black" />
				</View>
				<FieldContainer>
					<Field>
						<Label>Paciente:</Label>
						<FieldText>{appointment.paciente}</FieldText>
					</Field>

					<Field>
						<Label>Data e hora: </Label>
						<FieldText>{`${formatsDate(appointment.dataConsulta, 'weekday-date-time')}`}</FieldText>
					</Field>
				</FieldContainer>
			</Header>
				
			<Content>
				<Field>
					<Label>Observação: </Label>
					<FieldText>{appointment.observacao}</FieldText>
				</Field>
			</Content>
		</Container>
    )
}