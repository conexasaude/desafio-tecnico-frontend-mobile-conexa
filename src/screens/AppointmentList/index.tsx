import { FlatList, View } from 'react-native';
import { NativeStackScreenProps } from "@react-navigation/native-stack";
import { useEffect, useState } from 'react';
import Spinner from 'react-native-loading-spinner-overlay';
import { getAppointments } from '../../services/appointment';
import { Appointment } from '../../types/appointment';
import RootStackParamList from '../../types/rootStackParamList';
import AppointmentCard from '../../components/AppointmentCard';
import SearchInput from '../../components/SearchInput';
import { Container } from './styles';

type Props = NativeStackScreenProps<RootStackParamList, 'AppointmentList'>;

export default function AppointmentList({ navigation }: Props) {
	const [appointments, setAppointments] = useState<Appointment[]>([])
	const [loading, setLoading] = useState(false)
	const [search, setSearch] = useState("")

	async function fetchAppointments() {
		setLoading(true)
		const response = await getAppointments()
		setAppointments(response)
		setLoading(false)
	}

	async function navigateToDetails(appointment: Appointment) {
		navigation.navigate('AppointmentDetails', {
			appointment
		})
	}

	function filterAppointments(appointments: Appointment[]) {
		if(search !== '') {
			return appointments.filter(appointment => 
				String(appointment.id).toLocaleLowerCase().includes(search.toLocaleLowerCase())  || appointment.paciente.toLocaleLowerCase().includes(search.toLocaleLowerCase())
			)
		} else {
			return appointments
		}
	}
    
	useEffect(() => {
		fetchAppointments()
	}, [])

	return (
		<>
			<Spinner
				visible={loading}
				textContent={'Carregando...'}
				textStyle={{color: '#FFF'}}
			/>
				
			<Container>
				<SearchInput value={search} onChangeText={setSearch} />

				<FlatList
					data={filterAppointments(appointments)}
					renderItem={({item}) => 
						<AppointmentCard
							id={String(item.id)}
							onPress={() => navigateToDetails(item)} 
							pacient={item.paciente}
							date={item.dataConsulta}
						/>
					}
					style={{ paddingHorizontal: 16 }}
				>
				</FlatList>
			</Container>
		</>
	)
}