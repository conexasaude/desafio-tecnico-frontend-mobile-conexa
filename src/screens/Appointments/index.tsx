import React, { useEffect } from 'react';
import { useReduxSelector } from '@/hooks/useReduxSelector';
import { useReduxDispatch } from '@/hooks/useReduxDispatch';
import { FlatList } from 'react-native';
import { Card } from '@/components/Card';
import { useTheme } from '@emotion/react';
import { useNavigation } from '@react-navigation/native';
import { NavPropsLogged } from '@/routes/logged';
import { fetchAppointments } from '@/store/thunks/appointment.thunk';
import { useToast } from 'react-native-toast-notifications';
import * as Styled from './styles';

export function AppointmentsScreen() {
  const { show } = useToast();
  const { effects } = useTheme();
  const { navigate } = useNavigation<NavPropsLogged>();
  const { appointments } = useReduxSelector(state => state.appointment);

  const dispatch = useReduxDispatch();

  useEffect(() => {
    dispatch(fetchAppointments({ onError: show }));
  }, [show, dispatch]);

  return (
    <Styled.SafeArea>
      <Styled.Container>
        <Styled.Column>
          <Styled.Title>Painel de Consultas</Styled.Title>
          <Styled.Line />
        </Styled.Column>

        <Styled.Row>
          <Styled.DescriptionList>{`${appointments.length} consulta(s) agendadas`}</Styled.DescriptionList>
          <Styled.ScheduleButton
            title="Agendar Consulta"
            onPress={() => navigate('ScheduleAppointment')}
          />
        </Styled.Row>

        <FlatList
          data={appointments}
          keyExtractor={({ id }) => id.toString()}
          renderItem={({ item }) => (
            <Card
              testID="appointment-card"
              title={item.paciente}
              description={item.dataConsulta}
              button={{
                title: 'Ver Detalhes',
                onPress: () => navigate('DetailsAppointment', { id: item.id }),
              }}
            />
          )}
          contentContainerStyle={{ paddingHorizontal: effects.spacing.md }}
        />
      </Styled.Container>
    </Styled.SafeArea>
  );
}
