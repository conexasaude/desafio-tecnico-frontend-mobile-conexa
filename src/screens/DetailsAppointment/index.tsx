import React from 'react';
import * as Styled from './styles';

export function DetailsAppointmentScreen() {
  return (
    <Styled.Container>
      <Styled.Title>Details Appointment Screen</Styled.Title>
    </Styled.Container>
  );
}
