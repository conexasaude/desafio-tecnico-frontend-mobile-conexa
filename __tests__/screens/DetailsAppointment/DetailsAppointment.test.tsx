import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/lib/functions/Testing';

import { DetailsAppointmentScreen } from '@/screens/DetailsAppointment';

const mockDispatch = jest.fn();

const appointment = {
  id: 16,
  medico: {
    id: 1,
    nome: 'Dr. Alex Green',
    email: null,
  },
  paciente: 'Maria Brow',
  dataConsulta: '2020-04-23 08:30',
  observacao: 'Exemplo de consulta',
};

jest.mock('@/hooks/useReduxDispatch', () => ({
  useReduxDispatch: () => mockDispatch,
}));

jest.mock('@/hooks/useReduxSelector', () => ({
  useReduxSelector: () => ({ appointment }),
}));

describe('DetailsAppointments', () => {
  test('should render correctly', () => {
    renderWithThemeProvider(<DetailsAppointmentScreen />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<DetailsAppointmentScreen />);
  });

  test('should render all informations about appointments', () => {
    const { getByText } = renderWithThemeProvider(<DetailsAppointmentScreen />);
    expect(getByText('Maria Brow')).toBeDefined();
    expect(getByText('2020-04-23 08:30')).toBeDefined();
    expect(getByText('Exemplo de consulta')).toBeDefined();
  });
});
