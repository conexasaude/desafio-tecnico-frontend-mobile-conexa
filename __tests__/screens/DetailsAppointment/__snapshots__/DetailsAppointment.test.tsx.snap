// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsAppointments should matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
          "paddingHorizontal": 16,
          "paddingTop": 40,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#2E50D4",
            "fontFamily": "Nunito-Bold",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Nome do paciente:
    </Text>
    <Text
      style={
        [
          {
            "color": "#7F7F7F",
            "fontFamily": "Nunito-SemiBold",
            "fontSize": 14,
            "marginBottom": 16,
          },
          undefined,
        ]
      }
    >
      Maria Brow
    </Text>
    <Text
      style={
        [
          {
            "color": "#2E50D4",
            "fontFamily": "Nunito-Bold",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Data:
    </Text>
    <Text
      style={
        [
          {
            "color": "#7F7F7F",
            "fontFamily": "Nunito-SemiBold",
            "fontSize": 14,
            "marginBottom": 16,
          },
          undefined,
        ]
      }
    >
      2020-04-23 08:30
    </Text>
    <Text
      style={
        [
          {
            "color": "#2E50D4",
            "fontFamily": "Nunito-Bold",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Observação:
    </Text>
    <Text
      style={
        [
          {
            "color": "#7F7F7F",
            "fontFamily": "Nunito-SemiBold",
            "fontSize": 14,
            "marginBottom": 16,
          },
          undefined,
        ]
      }
    >
      Exemplo de consulta
    </Text>
  </View>
</RCTSafeAreaView>
`;
